### PR TITLE
SO-5916: Search term with ES synonym does not result in a match

### PR DESCRIPTION
**Pull request by adamfilep**

### CODE REVIEW

File: commons/com.b2international.index.tests/src/com/b2international/index/AnalyzerTest.java [hunk 1]
```diff
@@ -126,4 +166,21 @@ public void tokenizedWithSynonyms() throws Exception {
...
+	public void tokenizedWithSynonymsDefaultSearchAnalyzer() throws Exception {
+		DataWithTokenizedTextSearchAnalyzer bbq = new DataWithTokenizedTextSearchAnalyzer(KEY1, "bbq weekend");
+		DataWithTokenizedTextSearchAnalyzer stone = new DataWithTokenizedTextSearchAnalyzer(KEY2, "kidney stone");
+		indexDocuments(bbq, stone);
+		
+		// without specific synonym analyzer searches for synonyms return no hits
+		Hits<DataWithTokenizedTextSearchAnalyzer> hits = search(Query.select(DataWithTokenizedTextSearchAnalyzer.class)
+				.where(Expressions.dismax(
+					Expressions.matchTextAll("text.tokenized", "barbecue").withIgnoreStopwords(false),
+					Expressions.matchTextAll("text.tokenized", "calculus").withSynonyms(false)
```
Comment (cmark) [inline commons/com.b2international.index.tests/src/com/b2international/index/AnalyzerTest.java]: For consistency please disable stopwords explicitly.

File: commons/com.b2international.index/src/com/b2international/index/es/query/EsQueryBuilder.java [hunk 2]
```diff
@@ -330,6 +334,38 @@ private void visit(TextPredicate predicate) {
...
+			if (enableSynonym) {
+				searchAnalyzer = Analyzers.TOKENIZED_SYNONYMS;
+			} else {
+				searchAnalyzer = Analyzers.TOKENIZED;
+			}
+		}
+		
+		if (ignoreStopWords != null) {
+			if (ignoreStopWords) {
+				if (searchAnalyzer == null && searchAnalyzer == Analyzers.TOKENIZED) {
```
Comment (apeteri) [inline commons/com.b2international.index/src/com/b2international/index/es/query/EsQueryBuilder.java]: Impossible condition ⚠️

**Review by cmark — APPROVED** [review]
🥓

File: commons/com.b2international.index/src/com/b2international/index/query/TextPredicate.java [hunk 3]
```diff
@@ -81,6 +83,24 @@ public int prefixLength() {
 		return prefixLength;
 	}
 	
+	public Boolean synonym() {
+		return synonym;
+	}
+	
+	public TextPredicate withSynonyms(Boolean enableSynonyms) {
+		this.synonym = enableSynonyms;
```
Comment (apeteri) [inline commons/com.b2international.index/src/com/b2international/index/query/TextPredicate.java]: I'd suggest using the same name consistently (with-er method suffix, local parameter, field name), as occasionally we resort to method calls using reflection and a systematic naming convention helps a lot:

```suggestion
	public TextPredicate withSynonym(Boolean synonym) {
		this.synonym = synonym;
```

**Review by apeteri — CHANGES_REQUESTED** [review]
I've found two small items but otherwise changes look good!

File: commons/com.b2international.index/src/com/b2international/index/query/TextPredicate.java (hunk 3 above)
Comment (cmark) [reply commons/com.b2international.index/src/com/b2international/index/query/TextPredicate.java]: How about keeping the plural form here @apeteri and only changing the method argument name?

Comment (apeteri) [reply commons/com.b2international.index/src/com/b2international/index/query/TextPredicate.java]: The accessor and the field is also singular: https://github.com/b2ihealthcare/snow-owl/blob/48d0fa99fbed809fa96930f2d412436e1cc4e180/commons/com.b2international.index/src/com/b2international/index/query/TextPredicate.java#L86-L88

I'm not married to either form, whatever makes sense to you. 😺

Comment (cmark) [reply commons/com.b2international.index/src/com/b2international/index/query/TextPredicate.java]: `synonymsEnabled` ?

**Review by apeteri — APPROVED** [review]
👍

File: commons/com.b2international.index/src/com/b2international/index/query/TextPredicate.java (hunk 3 above)
Comment (apeteri) [reply commons/com.b2international.index/src/com/b2international/index/query/TextPredicate.java]: Sounds good!

**Review by nagyo — APPROVED** [review]
LGTM!

**Review by cmark — APPROVED** [review]
🥓

**Pull request merged**